device/sycl/src/seeding/doublet_counting.sycl

```diff
@@ -151,12 +161,13 @@ void doublet_counting(const seedfinder_config& config, sp_grid& internal_sp,
     // 1 dim ND Range for the kernel
     auto doubletCountNdRange = ::sycl::nd_range<1>{globalSize, localSize};
 
-    q->submit([&](::sycl::handler& h) {
-         DupletCount kernel(config, internal_sp_view,
-                            doublet_counter_container_view);
-         h.parallel_for<class doublet_count_kernel>(doubletCountNdRange,
-                                                    kernel);
-     }).wait_and_throw();
+    details::get_queue(queue)
+        .submit([&](::sycl::handler& h) {
+            DupletCount kernel(config, internal_sp_view,
+                               doublet_counter_container_view);
+            h.parallel_for<DupletCount>(doubletCountNdRange, kernel);
```

Comment:
Just, since it's a clean up, I realized later on that `parallel_for` doesn't need to be templated with the name of the kernel if they are created as functors...but I can remove that in some other PR.

Reply:
It's a oneAPI extension that you can forego specifying a template parameter to `parallel_for`. I think we should leave these be.

The change I made here is just to make use of the functor types that we have there already, instead of declaring types just inside of the function call itself. Which turns into a warning with [ComputeCpp](https://developer.codeplay.com/products/computecpp/ce/home/). (We discussed this either on Mattermost or on a ticket with @HadrienG2 a while ago.)

All in all, I think this part of the code should just stay like this. I'll have some other ideas for you to clean up, don't worry. :wink: